src/gpuarray_buffer_opencl.c

```diff
@@ -320,7 +320,64 @@ static const char CL_PREAMBLE[] =
   "#define load_half(p) vload_half(0, p)\n"
   "#define store_half(p, v) vstore_half_rtn(v, 0, p)\n"
   "#define GA_DECL_SHARED_PARAM(type, name) , __local type *name\n"
-  "#define GA_DECL_SHARED_BODY(type, name)\n";
+  "#define GA_DECL_SHARED_BODY(type, name)\n"
+  "#define acosf(x) acos(x)\n"
+  "#define acoshf(x) acosh(x)\n"
+  "#define asinf(x) asin(x)\n"
+  "#define asinhf(x) asinh(x)\n"
+  "#define atan2f(y, x) atan2(y, x)\n"
+  "#define atanf(x) atan(x)\n"
+  "#define atanhf(x) atanh(x)\n"
+  "#define cbrtf(x) cbrt(x)\n"
+  "#define ceilf(x) ceil(x)\n"
+  "#define copysignf(x, y) copysign(x, y)\n"
+  "#define cosf(x) cos(x)\n"
+  "#define coshf(x) cosh(x)\n"
+  "#define cospif(x) cospi(x)\n"
+  "#define erfcf(x) erfc(x)\n"
+  "#define erff(y) erf(x)\n"
+  "#define exp10f(x) exp10(x)\n"
+  "#define exp2f(x) exp2(x)\n"
+  "#define expf(x) exp(x)\n"
+  "#define expm1f(x) expm1(x)\n"
+  "#define fabsf(x) fabs(x)\n"
+  "#define fdimf(x, y) fdim(x, y)\n"
+  "#define fdividef(x, y) (x/y)\n"
+  "#define floorf(x) floor(x)\n"
+  "#define fmaf(x, y, z) fma(x, y, z)\n"
+  "#define fmaxf(x, y) fmax(x, y)\n"
+  "#define fminf(x, y) fmin(x, y)\n"
+  "#define fmodf(x, y) fmod(x, y)\n"
+  "#define frexpf(x, nptr) frexp(x, nptr)\n"
+  "#define hypotf(x, y) hypot(x, y)\n"
+  "#define ilogbf(x) ilogb(x)\n"
+  "#define ldexpf(x, exp) ldexp(x, exp)\n"
+  "#define lgammaf(x) lgamma(x)\n"
+  "#define log10f(x) log10(x)\n"
+  "#define log1pf(x) log1p(x)\n"
+  "#define log2f(x) log2(x)\n"
+  "#define logbf(x) logb(x)\n"
+  "#define logf(x) log(x)\n"
+  "#define modff(x, iptr) modf(x, iptr)\n"
+  "#define nanf(tagp) NAN\n"
+  "#define nan(tagp) NAN\n"
+  "#define nextafterf(x, y) nextafter(x, y)\n"
+  "#define powf(x, y) pow(x, y)\n"
+  "#define remainderf(x, y) remainder(x, y)\n"
+  "#define remquof(x, y, quo) remquo(x, y, quo)\n"
+  "#define rhypotf(x, y) (1/hypot(x, y))\n"
```

Comment:
This might not respect tolerance specifications.

Again I prefer unimplemented to wrong.